airbyte-webapp/src/views/Connector/ConnectorDocumentationLayout/ResourceNotAvailable/index.tsx

```diff
@@ -0,0 +1,55 @@
+import { faArrowDown } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { FormattedMessage } from "react-intl";
+
+import { Button } from "components/ui/Button";
+import { Text } from "components/ui/Text";
+
+import { SourceDefinitionRead } from "core/request/AirbyteClient";
+import { useDocumentationPanelContext } from "views/Connector/ConnectorDocumentationLayout/DocumentationPanelContext";
+
+import styles from "./ResourceNotAvailable.module.scss";
+import octaviaWorker from "../octavia-worker.png";
+import { useAnalyticsTrackFunctions } from "../useAnalyticsTrackFunctions";
+
+export const ResourceNotAvailable: React.FC<
+  React.PropsWithChildren<{
+    activeTab: "erd" | "schema";
+    isRequested: boolean;
+    setRequested: (val: boolean) => void;
+  }>
+> = ({ activeTab, setRequested, isRequested }) => {
+  const { selectedConnectorDefinition } = useDocumentationPanelContext();
+  const { trackRequest } = useAnalyticsTrackFunctions();
+
+  return (
+    <div className={styles.requestContainer}>
+      <img src={octaviaWorker} alt="" />
+      {isRequested ? (
+        <Text size="lg">
+          <FormattedMessage id="sources.request.thankYou" />
+        </Text>
+      ) : (
+        <>
+          <Text size="lg">
+            <FormattedMessage id="sources.request.prioritize" />
+          </Text>
+          <FontAwesomeIcon icon={faArrowDown} />
+          <Button
+            variant="primary"
+            onClick={() => {
+              trackRequest({
+                sourceDefinitionId: (selectedConnectorDefinition as SourceDefinitionRead).sourceDefinitionId,
+                connectorName: selectedConnectorDefinition.name,
+                requestType: activeTab,
+              });
+              setRequested(true);
+            }}
```

Comment:
```suggestion
            onClick={() => {
              if (isSourceDefinition(selectedConnectorDefinition)) {
                trackRequest({
                  sourceDefinitionId: selectedConnectorDefinition.sourceDefinitionId,
                  connectorName: selectedConnectorDefinition.name,
                  requestType: activeTab,
                });
              }
              setRequested(true);
            }}
```

That way we don't have manual type casts here, that might break our code if someone accidentally would use this component while a destination connector is open, and thus not realy on implicit knowledge and flow of a completely other component.